hazelcast/src/main/java/com/hazelcast/spi/discovery/impl/DefaultDiscoveryService.java

```diff
@@ -155,11 +155,17 @@ private boolean validateCandidate(DiscoveryNode candidate) {
 
             List<DiscoveryStrategy> discoveryStrategies = new ArrayList<DiscoveryStrategy>();
             for (DiscoveryStrategyFactory factory : factories) {
+                //as a side-effect the buildDiscoveryStrategy() removes a config of the created strategy
                 DiscoveryStrategy discoveryStrategy = buildDiscoveryStrategy(factory, discoveryStrategyConfigs);
                 if (discoveryStrategy != null) {
                     discoveryStrategies.add(discoveryStrategy);
                 }
             }
+            if (!discoveryStrategyConfigs.isEmpty()) {
```

Comment:
I agree with David. It isn't nice if an object gets modified in a verification method.